### DISCUSSION TITLE
Using prefix on APIRouter with websockets doesn't work

### DISCUSSION THREAD

**Discussion by danrobinson88** [fastapi/fastapi#6955] [Questions] [answered]
Hi,

to reproduce:

```
router = APIRouter(
    prefix="/v1"
)

@router.websocket_route("/calculate")
async def websocket_endpoint(websocket: WebSocket) -> None:
    pass

app = FastAPI()
app.include_router(router)
```

This doesn't work, meaning calling the server with websocket with path /v1/calculate fail.

Whereas, it does work without prefix:

```
router = APIRouter()

@router.websocket_route("/v1/calculate")
async def websocket_endpoint(websocket: WebSocket) -> None:
    pass

app = FastAPI()
app.include_router(router)
```

fastapi version: 0.63.0

Fixing this would be highly appreciated :)

**Comment by Kludex** [comment, marked as answer]
Solution implemented on: https://github.com/tiangolo/fastapi/pull/2640

**Comment by tiangolo** [comment]
Thanks for the help here @Kludex ! 👏  🙇 

If that solves the original problem, then you can close this issue @danrobinson88  ✔️

> Sorry for the long delay! 🙈  I wanted to personally address each issue/PR and they piled up through time, but now I'm checking each one in order.

**Comment by danrobinson88** [comment]
@tiangolo it does, will do, thanks.
